renderer/components/action-bar/controls/advanced.js

```diff
@@ -0,0 +1,204 @@
+
+// Containers
+import {connect, ActionBarContainer, CropperContainer} from '../../../containers';
+
+// Utilities
+import {handleWidthInput, handleHeightInput} from '../../../utils/inputs';
+
+const advancedStyles = css`
+  .advanced {
+    height: 50px;
+    display: flex;
+    flex: 1;
+    align-items: center;
+    justify-content: space-between;
+  }
+`;
+
+const AdvancedControls = {};
+
+class Left extends React.Component {
+  render() {
+    const {toggleAdvanced, toggleRatioLock, ratioLocked, ratio = []} = this.props;
+
+    return (
+      <div className="advanced">
+        <BackIcon onClick={toggleAdvanced}/>
+        <div className="select">
+          <span>{ratio[0]}:{ratio[1]}</span>
+          <DropdownArrowIcon size={15}/>
+        </div>
+        <LinkIcon active={ratioLocked} onClick={() => toggleRatioLock()}/>
+        <style jsx>{advancedStyles}</style>
+        <style jsx>{`
+          .select {
+            border: 1px solid #ddd;
+            border-radius: 4px;
+            font-size: 0.7rem;
+            height: 1.1rem;
+            transition: border 0.12s ease-in-out;
+            display: flex;
+            align-items: center;
+            padding: 0 5px;
+          }
+
+          .select span {
+            flex: 1;
+            text-align: right;
+            padding-right: 5px;
+          }
+
+          .select:hover {
+            border-color: #ccc;
+          }
+        `}</style>
+      </div>
+    );
+  }
+}
+
+Left.propTypes = {
+  toggleAdvanced: PropTypes.func.isRequired,
+  toggleRatioLock: PropTypes.func.isRequired,
+  ratioLocked: PropTypes.bool,
+  ratio: PropTypes.array
+};
+
+AdvancedControls.Left = connect(
+  [ActionBarContainer, CropperContainer],
+  ({ratioLocked}, {ratio}) => ({ratio, ratioLocked}),
+  ({toggleAdvanced, toggleRatioLock}) => ({toggleAdvanced, toggleRatioLock})
+)(Left);
+
+class Right extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      width: props.width,
+      height: props.height
+    };
+  }
+
+  componentWillReceiveProps(newProps) {
+    this.setState({
+      width: newProps.width,
+      height: newProps.height
+    });
+  }
+
+  onWidthChange = e => {
+    const {x, y, setBounds, ratioLocked, ratio} = this.props;
+    const {value} = e.target;
+    const {heightInput, widthInput} = this;
+
+    this.setState({width: value});
+    handleWidthInput({x, y, setBounds, ratioLocked, ratio, value, widthInput, heightInput});
+  }
+
+  onHeightChange = e => {
+    const {x, y, setBounds, ratioLocked, ratio} = this.props;
+    const {value} = e.target;
+    const {heightInput, widthInput} = this;
+
+    this.setState({height: value});
+    handleHeightInput({x, y, setBounds, ratioLocked, ratio, value, widthInput, heightInput});
+  }
+
+  render() {
+    const {width, height} = this.state;
+
+    return (
+      <div className="advanced">
+        <div className="dimensions">
+          <input
+            ref={
+              c => {
```

Comment:
Use the `createRef` API: https://reactjs.org/blog/2018/03/29/react-v-16-3.html#createref-api

Reply:
I tried using that at first, but for some reason, it wasn't working. I'll try again

Reply:
You have to upgrade `react` for it to work.